src/langchain_google_spanner/vector_store.py

```diff
@@ -0,0 +1,1181 @@
+
+        if not database.exists():
+            raise Exception("Database with id-{} doesn't exist.".format(database_id))
+
+        database.reload()
+
+        ddl = SpannerVectorStore._generate_sql(
+            database.database_dialect,
+            table_name,
+            id_column,
+            content_column,
+            embedding_column,
+            metadata_columns,
+            primary_key,
+            secondary_indexes
+        )
+
+        operation = database.update_ddl(ddl)
+
+        print("Waiting for operation to complete...")
+        operation.result(100000)
+
+        return True
+
+    @staticmethod
+    def _generate_sql(
+        dialect,
+        table_name,
+        id_column,
+        content_column,
+        embedding_column,
+        column_configs,
+        primary_key,
+        secondary_indexes: Optional[List[SecondaryIndex]] = None,
+    ):
+        """
+        Generate SQL for creating the vector store table.
+
+        Parameters:
+        - dialect: The database dialect.
+        - table_name: The name of the table.
+        - id_column: The name of the row ID column.
+        - content_column: The name of the content column.
+        - embedding_column: The name of the embedding column.
+        - column_names: List of tuples containing metadata column information.
+
+        Returns:
+        - str: The generated SQL.
+        """
+        create_table_statement = f"CREATE TABLE {table_name} (\n"
+
+        if not isinstance(id_column, TableColumn):
+            if dialect == DatabaseDialect.POSTGRESQL:
+                id_column = TableColumn(id_column, "varchar(36)", is_null=True)
+            else:
+                id_column = TableColumn(id_column, "STRING(36)", is_null=True)
+
+        if not isinstance(content_column, TableColumn):
+            if dialect == DatabaseDialect.POSTGRESQL:
+                content_column = TableColumn(content_column, "text", is_null=True)
+            else:
+                content_column = TableColumn(
+                    content_column, "STRING(MAX)", is_null=True
+                )
+
+        if not isinstance(embedding_column, TableColumn):
+            if dialect == DatabaseDialect.POSTGRESQL:
+                embedding_column = TableColumn(
+                    embedding_column, "float8[]", is_null=True
+                )
+            else:
+                embedding_column = TableColumn(
+                    embedding_column, "ARRAY<FLOAT64>", is_null=True
+                )
+
+        configs = [id_column, content_column, embedding_column]
+        configs.extend(column_configs)
+        column_configs = configs
+
+        if primary_key is None:
+            primary_key = id_column.name + "," + content_column.name
+
+        if column_configs is not None:
+            for column_config in column_configs:
+                # Append column name and data type
+                column_sql = f"  {column_config.name} {column_config.type}"
```

Comment:
Should column names also be in quotes?

Reply:
it's not required as Spanner doesn't support hyphens.